server/src/main/scala/org/http4s/server/middleware/ChunkAggregator.scala

```diff
@@ -1,38 +1,20 @@
-/* TODO fs2 port what replaces unemit?
 package org.http4s
 package server
 package middleware
 
-import scala.annotation.tailrec
-
 import fs2._
+import fs2.interop.cats._
 import org.http4s.headers._
 
 object ChunkAggregator {
-
-  @tailrec
-  private[ChunkAggregator] def reduce(acc: Chunk[Byte], chunks: Seq[Chunk[Byte]]): List[Chunk[Byte]] = {
-    if (chunks.tail.nonEmpty) reduce(acc ++ chunks.head, chunks.tail)
-    else (acc ++ chunks.head) :: Nil
-  }
-
-  private[ChunkAggregator] def compact(body: EntityBody): List[ByteVector] = {
-    val (chunks, tail) = body.unemit
-    if (chunks.nonEmpty &&
-      tail.isInstanceOf[Halt] &&
-      (tail.asInstanceOf[Halt].cause eq End)) reduce(ByteVector.empty, chunks)
-    else Nil
-  }
-
-  def apply(service: HttpService): HttpService = service.map {
+  def apply(service: HttpService): HttpService = service.flatMapF[MaybeResponse] {
     case response: Response =>
-      val chunks = compact(response.body)
-      if (chunks.nonEmpty) {
-        val h = response.headers.put(`Content-Length`(chunks.head.length))
-        response.copy(body = emitAll(chunks), headers = h)
+      response.body.runLog.map { bytes =>
+        if (bytes.nonEmpty) {
+          val h = response.headers.put(`Content-Length`(bytes.length.toLong))
```

Comment:
If we're going to force a `Content-Length` header, we should also ensure there isn't a `Transfer-Encoding: chunked` header present.